drivers/sensor/iis2dlpc/iis2dlpc.h

```diff
@@ -0,0 +1,152 @@
+
+union axis3bit16_t {
+	s16_t i16bit[3];
+	u8_t u8bit[6];
+};
+
+#if defined(CONFIG_IIS2DLPC_ODR_1_6)
+	#define IIS2DLPC_DEFAULT_ODR	IIS2DLPC_XL_ODR_1Hz6_LP_ONLY
+#elif defined(CONFIG_IIS2DLPC_ODR_12_5)
+	#define IIS2DLPC_DEFAULT_ODR	IIS2DLPC_XL_ODR_12Hz5
+#elif defined(CONFIG_IIS2DLPC_ODR_25)
+	#define IIS2DLPC_DEFAULT_ODR	IIS2DLPC_XL_ODR_25Hz
+#elif defined(CONFIG_IIS2DLPC_ODR_50)
+	#define IIS2DLPC_DEFAULT_ODR	IIS2DLPC_XL_ODR_50Hz
+#elif defined(CONFIG_IIS2DLPC_ODR_100) || \
+	defined(CONFIG_IIS2DLPC_ODR_RUNTIME)
+	#define IIS2DLPC_DEFAULT_ODR	IIS2DLPC_XL_ODR_100Hz
+#elif defined(CONFIG_IIS2DLPC_ODR_200)
+	#define IIS2DLPC_DEFAULT_ODR	IIS2DLPC_XL_ODR_200Hz
+#elif defined(CONFIG_IIS2DLPC_ODR_400)
+	#define IIS2DLPC_DEFAULT_ODR	IIS2DLPC_XL_ODR_400Hz
+#elif defined(CONFIG_IIS2DLPC_ODR_800)
+	#define IIS2DLPC_DEFAULT_ODR	IIS2DLPC_XL_ODR_800Hz
+#elif defined(CONFIG_IIS2DLPC_ODR_1600)
+	#define IIS2DLPC_DEFAULT_ODR	IIS2DLPC_XL_ODR_1k6Hz
+#endif
+
+/* Return ODR reg value based on data rate set */
+#define IIS2DLPC_ODR_TO_REG(_odr) \
+	((_odr <= 1) ? IIS2DLPC_XL_ODR_1Hz6_LP_ONLY : \
+	 (_odr <= 12) ? IIS2DLPC_XL_ODR_12Hz5 : \
+	 ((31 - __builtin_clz(_odr / 25))) + 3)
+
+/* FS reg value from Full Scale */
+#define IIS2DLPC_FS_TO_REG(_fs)	(30 - __builtin_clz(_fs))
+
+#if defined(CONFIG_IIS2DLPC_ACCEL_RANGE_RUNTIME) || \
+	defined(CONFIG_IIS2DLPC_ACCEL_RANGE_2G)
+	#define IIS2DLPC_ACC_FS		IIS2DLPC_2g
+#elif defined(CONFIG_IIS2DLPC_ACCEL_RANGE_4G)
+	#define IIS2DLPC_ACC_FS		IIS2DLPC_4g
+#elif defined(CONFIG_IIS2DLPC_ACCEL_RANGE_8G)
+	#define IIS2DLPC_ACC_FS		IIS2DLPC_8g
+#elif defined(CONFIG_IIS2DLPC_ACCEL_RANGE_16G)
+	#define IIS2DLPC_ACC_FS		IIS2DLPC_16g
+#endif
+
+/* Acc Gain value in ug/LSB in High Perf mode */
+#define IIS2DLPC_FS_2G_GAIN		244
+#define IIS2DLPC_FS_4G_GAIN		488
+#define IIS2DLPC_FS_8G_GAIN		976
+#define IIS2DLPC_FS_16G_GAIN		1952
+
+#define IIS2DLPC_SHFT_GAIN_NOLP1	2
+#define IIS2DLPC_ACCEL_GAIN_DEFAULT_VAL IIS2DLPC_FS_2G_GAIN
+#define IIS2DLPC_FS_TO_GAIN(_fs, _lp1) \
+		(IIS2DLPC_FS_2G_GAIN << ((_fs) + (_lp1)))
+
+/* shift value for power mode */
+#define IIS2DLPC_SHIFT_PM1		4
+#define IIS2DLPC_SHIFT_PMOTHER		2
+
+/**
+ * struct iis2dlpc_device_config - iis2dlpc hw configuration
+ * @bus_name: Pointer to bus master identifier.
+ * @pm: Power mode (lis2dh_powermode).
+ * @int_gpio_port: Pointer to GPIO PORT identifier.
+ * @int_gpio_pin: GPIO pin number connecter to sensor int pin.
+ * @int_pin: Sensor int pin (int1/int2).
+ */
+struct iis2dlpc_device_config {
+	const char *bus_name;
+	iis2dlpc_mode_t pm;
+#ifdef CONFIG_IIS2DLPC_TRIGGER
+	const char *int_gpio_port;
+	u8_t int_gpio_pin;
+	u8_t int_gpio_flags;
+	u8_t int_pin;
+#ifdef CONFIG_IIS2DLPC_PULSE
+	u8_t pulse_trigger;
+	u8_t pulse_ths[3];
+	u8_t pulse_shock;
+	u8_t pulse_ltncy;
+	u8_t pulse_quiet;
+#endif /* CONFIG_IIS2DLPC_PULSE */
+#endif /* CONFIG_IIS2DLPC_TRIGGER */
+};
+
+/* sensor data forward declaration (member definition is below) */
+struct iis2dlpc_data;
```

Comment:
Is this actually necessary?

Reply:
Of course not. It probably remained there by mistake.